lib/proxy_ssh/proxy.go

```diff
@@ -254,18 +258,54 @@ func (p *ProxyAccess) passwordCallback(conn ssh.ConnMetadata, pass []byte) (*ssh
 	return nil, fmt.Errorf("invalid access")
 }
 
-func Init(fish *fish.Fish, address string) error {
-	// TODO: cannot find a way to _not_ do this.
-	// TODO: ref cmd, we do not need to add config entry
-	// ref: https://gist.github.com/devinodaniel/8f9b8a4f31573f428f29ec0e884e6673
-	privateBytes, err := os.ReadFile("id_rsa")
+func Init(fish *fish.Fish, id_rsa_path string, address string) error {
+	// First, try and read the file if it exists already.  Otherwise, it is the
+	// first execution, generate the private / public keys.  The SSH server
+	// requires at least one identity loaded to run.
+	privateBytes, err := os.ReadFile(id_rsa_path)
 	if err != nil {
-		return fmt.Errorf("proxy_ssh: failed to load private key: %v", err)
+		// If it cannot be loaded, this is the first execution, generate it.
+		log.Infof("SSH Proxy: could not load %q, generating now.", id_rsa_path)
+		rsaKey, err := rsa.GenerateKey(rand.Reader, 4096)
+		if err != nil {
+			return fmt.Errorf("proxy_ssh: could not generate private key %q: %w", id_rsa_path, err)
+		}
+		pemKey := pem.EncodeToMemory(
+			&pem.Block{
+				Type:  "RSA PRIVATE KEY",
+				Bytes: x509.MarshalPKCS1PrivateKey(rsaKey),
+			},
+		)
+		// The public key isn't really needed for anything currently, however
+		// generating the public counterpart may as well be done now.
+		pubKey := pem.EncodeToMemory(
```

Comment:
Not quite sure I understand why we need pub key - maybe you have something in mind?

In case it's just for fun - let's not do extra generation/storing of the ssh pub key - [KISS](https://en.wikipedia.org/wiki/KISS_principle) is our way. Everyone who will need a pub key - can generate it using openssh tools from the private key and we will not need to support this interface forever. If we will need it in the future - we can easily add this logic back.